### DISCUSSION TITLE
Severe performance problems after enabling reverse proxy and HTTPS

### DISCUSSION THREAD

**Discussion by marian001** [Budibase/budibase#7908] [Help] [answered]
Hi all,

I started experiencing significant decline in performance when designing my application. This started happening soon after I successfully managed to set up subdomain redirection, reverse proxy (nginx), HTTPS (certbot) and user authentication through Auth0.

I'm running on DigitalOcean 8 GB Memory / 4 Intel vCPUs and my application export is only around 300kB, so neither of these seem to be the problem.

After looking into browser dev tools > Network > Console, it looks like there are indications of what's causing the problems:

`WebSocket connection to 'wss://aaaaaa.bbbbbb.cc/socket/client/?EIO=4&transport=websocket&sid=QUzsu4oIP07bZ0_fABO7' failed: 
doOpen @ budibase-client.js?v=1.4.7:47
open @ budibase-client.js?v=1.4.7:47
probe @ budibase-client.js?v=1.4.7:47
onOpen @ budibase-client.js?v=1.4.7:47
onHandshake @ budibase-client.js?v=1.4.7:47
onPacket @ budibase-client.js?v=1.4.7:47
oU.emit @ budibase-client.js?v=1.4.7:47
onPacket @ budibase-client.js?v=1.4.7:47
(anonymous) @ budibase-client.js?v=1.4.7:47
onData @ budibase-client.js?v=1.4.7:47
oU.emit @ budibase-client.js?v=1.4.7:47
onLoad @ budibase-client.js?v=1.4.7:47
t.onreadystatechange @ budibase-client.js?v=1.4.7:47
`

`POST https://aaaaaa.bbbbbb.cc/socket/client/?EIO=4&transport=polling&t=ODbW201&sid=OiQGdqHsv40de2DTABOg 400 (Bad Request)
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doWrite	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
Kq	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
write	@	budibase-client.js?v=1.4.7:47
send	@	budibase-client.js?v=1.4.7:47
flush	@	budibase-client.js?v=1.4.7:47
sendPacket	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
doOpen	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
IU	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
setTimeout (async)		
reconnect	@	budibase-client.js?v=1.4.7:47
onclose	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onClose	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
setTimeout (async)		
resetPingTimeout	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
doOpen	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
IU	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
setTimeout (async)		
reconnect	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47`

`GET https://aaaaaa.bbbbbb.cc/socket/client/?EIO=4&transport=polling&t=ODbW202&sid=OiQGdqHsv40de2DTABOg 400 (Bad Request)
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
doOpen	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
IU	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
setTimeout (async)		
reconnect	@	budibase-client.js?v=1.4.7:47
onclose	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onClose	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
setTimeout (async)		
resetPingTimeout	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
doOpen	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
IU	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
setTimeout (async)		
reconnect	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47`

There are thousands of these error after being on Design page for some time. This is the same on any browser..  and I also tried to import the application into new application instance (on the same server) and that didn't help either.

Maybe one more thing...  I tried to install Budibase CLI (on top of DO Budibase image) based on Recipes because I thought it would add CLI functionality, but it seemed to try to install the whole Budibase...   so I'm not sure if I didn't mess something up.

I don't have much experience with networking setups, so I'd like to ask for help in diagnosing this further and resolving the problem.

Thanks

p.s. Outside of this, I think I managed to find all we need within Budibase for our portal, so I can't wait to start building...

**Comment by aptkingston** [comment, marked as answer]
Hey @marian001.

There are a couple of issues here. The first is that the websocket isn't connecting, which is probably because of your second layer of reverse proxy in front of Budibase. To fix that, you'll need to add a location block to your nginx config as follows:
```
    location /socket/ {
      proxy_http_version  1.1;
      proxy_set_header    Upgrade     $http_upgrade;
      proxy_set_header    Connection  'upgrade';
      proxy_set_header    Host        $host;
      proxy_cache_bypass  $http_upgrade;
      proxy_pass          http://localhost:10000; # Change to your upstream URL
    }
```

The second issue is that we are polling too agressively to attempt to reconnect the websocket whenever it fails to connect. To address that, I've opened a PR here: https://github.com/Budibase/budibase/pull/7909 which will fix this performance degredation you're seeing. We'll roll out this very soon.

So to fix your issue you can either add that nginx config (which will allow the websocket to work) or you can wait until we deploy our fix for the polling (which will leave the websocket broken, which is fine, but stop you seeing any issues because of it).

**Reply by rpmcoding** [reply]
I was having the same issue.  Thanks for the help.

**Reply by marian001** [reply]
Great! That resolved the problem. Thank you, @aptkingston 

This should be added to this part of documentation: https://docs.budibase.com/docs/reverse-proxy

I still have some errors in console although these seem to be unrelated to the previous problem. And these are not accumulating with time, so also not related to reconnection attempts. Please, let me know if these need further attention:

`Client received invalid JSON
receiveMessage	@	preview:52
postMessage (async)		
_postMessage	@	inpage.js:1
_write	@	inpage.js:1
w	@	inpage.js:1
_	@	inpage.js:1
y.uncork	@	inpage.js:1
_onData	@	inpage.js:1
_onMessage	@	inpage.js:1
postMessage (async)		
_postMessage	@	contentscript.js:1
_write	@	contentscript.js:1
_onData	@	contentscript.js:1
_onMessage	@	contentscript.js:1
postMessage (async)		
_postMessage	@	inpage.js:1
_write	@	inpage.js:1
_handshake	@	inpage.js:1
i	@	inpage.js:1
(anonymous)	@	inpage.js:1
(anonymous)	@	inpage.js:1
1.@babel/runtime/helpers/interopRequireDefault	@	inpage.js:1
i	@	inpage.js:1
e	@	inpage.js:1
(anonymous)	@	inpage.js:1
(anonymous)	@	contentscript.js:1
1.../../shared/modules/mv3.utils	@	contentscript.js:1
i	@	contentscript.js:1
e	@	contentscript.js:1
(anonymous)	@	contentscript.js:1`

`preview:52 Client received invalid JSON
receiveMessage	@	preview:52
postMessage (async)		
_postMessage	@	inpage.js:1
_write	@	inpage.js:1
w	@	inpage.js:1
(anonymous)	@	inpage.js:1
y.write	@	inpage.js:1
g	@	inpage.js:17
f	@	inpage.js:8
o.emit	@	inpage.js:8
_	@	inpage.js:17
v	@	inpage.js:17
w.push	@	inpage.js:17
_write	@	inpage.js:1
w	@	inpage.js:17
(anonymous)	@	inpage.js:17
y.write	@	inpage.js:17
g	@	inpage.js:17
f	@	inpage.js:8
o.emit	@	inpage.js:8
_	@	inpage.js:17
v	@	inpage.js:17
w.push	@	inpage.js:17
middleware	@	inpage.js:1
(anonymous)	@	inpage.js:17
_runMiddleware	@	inpage.js:17
_runAllMiddleware	@	inpage.js:17
await in _runAllMiddleware (async)		
_processRequest	@	inpage.js:17
_handle	@	inpage.js:17
handle	@	inpage.js:17
r.sendSiteMetadata	@	inpage.js:1
await in r.sendSiteMetadata (async)		
e	@	inpage.js:1`

`Client received invalid JSON
receiveMessage	@	preview:52
postMessage (async)		
_postMessage	@	contentscript.js:1
_write	@	contentscript.js:1
w	@	contentscript.js:1
(anonymous)	@	contentscript.js:1
y.write	@	contentscript.js:1
g	@	contentscript.js:17
d	@	contentscript.js:8
o.emit	@	contentscript.js:8
_	@	contentscript.js:17
v	@	contentscript.js:17
w.push	@	contentscript.js:17
_write	@	contentscript.js:17
w	@	contentscript.js:17
(anonymous)	@	contentscript.js:17
y.write	@	contentscript.js:17
g	@	contentscript.js:17
d	@	contentscript.js:8
o.emit	@	contentscript.js:8
_	@	contentscript.js:17
v	@	contentscript.js:17
w.push	@	contentscript.js:17
_write	@	contentscript.js:17
w	@	contentscript.js:17
(anonymous)	@	contentscript.js:17
y.write	@	contentscript.js:17
g	@	contentscript.js:17
d	@	contentscript.js:8
o.emit	@	contentscript.js:8
_	@	contentscript.js:17
v	@	contentscript.js:17
w.push	@	contentscript.js:17
_onMessage	@	contentscript.js:8
(anonymous)	@	contentscript.js:8`

**Reply by aptkingston** [reply]
Happy to hear so! `Client received invalid JSON` is usally caused by a browser extension of some sort... `contentscript.js` is not part of our codebase, so I'm assuming something else is injecting these messages. As you've noticed, these won't cause any issues because we deliberately ignore events that didn't originate from our own code.

**Reply by marian001** [reply]
When publishing an application, I noticed some delay, although not as bad. So I checked the console again and found a few more error types:

`DevTools failed to load source map: Could not load content for chrome-extension://neooppigbkahgfdhbpbhcccgpimeaafi/static/js/content.js.map: System error: net::ERR_BLOCKED_BY_CLIENT`

`budibase-client.js?v=1.4.7:47 WebSocket connection to 'wss://aaaaa.bbbbb.cc/socket/client/?EIO=4&transport=websocket&sid=YBZAmML-3SFtva8AAC8s' failed: Insufficient resources
doOpen	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
probe	@	budibase-client.js?v=1.4.7:47
onOpen	@	budibase-client.js?v=1.4.7:47
onHandshake	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
doOpen	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
IU	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
tW	@	budibase-client.js?v=1.4.7:47
nW	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
window.loadBudibase	@	budibase-client.js?v=1.4.7:47
receiveMessage	@	preview:92
postMessage (async)		
V	@	index.77ea7459.js:922
n.$$.update	@	index.77ea7459.js:922
CZ	@	index.77ea7459.js:4
Dw	@	index.77ea7459.js:4
Promise.then (async)		
gU	@	index.77ea7459.js:4
LZ	@	index.77ea7459.js:4
(anonymous)	@	index.77ea7459.js:4
(anonymous)	@	index.77ea7459.js:4
i	@	index.77ea7459.js:4
r	@	index.77ea7459.js:4
pushstate	@	index.77ea7459.js:4
history.<computed>	@	index.77ea7459.js:4
await in history.<computed> (async)		
(anonymous)	@	index.77ea7459.js:4
h	@	index.77ea7459.js:922
(anonymous)	@	index.77ea7459.js:4
On	@	index.77ea7459.js:4
l	@	index.77ea7459.js:343`

`POST https://aaaaa.bbbbb.cc/socket/client/?EIO=4&transport=polling&t=ODcywuj&sid=LWjySzBaBAZrkmx0AC8x 400 (Bad Request)
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doWrite	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
Kq	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
write	@	budibase-client.js?v=1.4.7:47
send	@	budibase-client.js?v=1.4.7:47
flush	@	budibase-client.js?v=1.4.7:47
sendPacket	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onPacket	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
doOpen	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
IU	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
tW	@	budibase-client.js?v=1.4.7:47
nW	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
window.loadBudibase	@	budibase-client.js?v=1.4.7:47
receiveMessage	@	preview:92
postMessage (async)		
V	@	index.77ea7459.js:922
n.$$.update	@	index.77ea7459.js:922
CZ	@	index.77ea7459.js:4
Dw	@	index.77ea7459.js:4
Promise.then (async)		
gU	@	index.77ea7459.js:4
LZ	@	index.77ea7459.js:4
(anonymous)	@	index.77ea7459.js:4
(anonymous)	@	index.77ea7459.js:4
i	@	index.77ea7459.js:4
r	@	index.77ea7459.js:4
pushstate	@	index.77ea7459.js:4
history.<computed>	@	index.77ea7459.js:4
await in history.<computed> (async)		
(anonymous)	@	index.77ea7459.js:4
X	@	index.77ea7459.js:922
x	@	index.77ea7459.js:922
postMessage (async)		
SM	@	budibase-client.js?v=1.4.7:22
selectComponent	@	budibase-client.js?v=1.4.7:22
a	@	budibase-client.js?v=1.4.7:47`

`GET https://aaaaa.bbbbb.cc/socket/client/?EIO=4&transport=polling&t=ODcywuk&sid=LWjySzBaBAZrkmx0AC8x 400 (Bad Request)
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
onData	@	budibase-client.js?v=1.4.7:47
oU.emit	@	budibase-client.js?v=1.4.7:47
onLoad	@	budibase-client.js?v=1.4.7:47
t.onreadystatechange	@	budibase-client.js?v=1.4.7:47
XMLHttpRequest.send (async)		
create	@	budibase-client.js?v=1.4.7:47
SU	@	budibase-client.js?v=1.4.7:47
request	@	budibase-client.js?v=1.4.7:47
doPoll	@	budibase-client.js?v=1.4.7:47
poll	@	budibase-client.js?v=1.4.7:47
doOpen	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
IU	@	budibase-client.js?v=1.4.7:47
open	@	budibase-client.js?v=1.4.7:47
tW	@	budibase-client.js?v=1.4.7:47
nW	@	budibase-client.js?v=1.4.7:47
(anonymous)	@	budibase-client.js?v=1.4.7:47
window.loadBudibase	@	budibase-client.js?v=1.4.7:47
receiveMessage	@	preview:92
postMessage (async)		
V	@	index.77ea7459.js:922
n.$$.update	@	index.77ea7459.js:922
CZ	@	index.77ea7459.js:4
Dw	@	index.77ea7459.js:4
Promise.then (async)		
gU	@	index.77ea7459.js:4
LZ	@	index.77ea7459.js:4
(anonymous)	@	index.77ea7459.js:4
(anonymous)	@	index.77ea7459.js:4
i	@	index.77ea7459.js:4
r	@	index.77ea7459.js:4
pushstate	@	index.77ea7459.js:4
history.<computed>	@	index.77ea7459.js:4
await in history.<computed> (async)		
(anonymous)	@	index.77ea7459.js:4
X	@	index.77ea7459.js:922
x	@	index.77ea7459.js:922
postMessage (async)		
SM	@	budibase-client.js?v=1.4.7:22
selectComponent	@	budibase-client.js?v=1.4.7:22
a	@	budibase-client.js?v=1.4.7:47`

`Failed to load resource: the server responded with a status of 400 (Bad Request)
/socket/client/?EIO=4&transport=polling&t=ODd327X&sid=yhX3YoccsUUM1zOjADFS:1 `

Number of these is growing...   
For completeness I'm adding the contents of my /etc/nginx/sites-available/default file:


```
# Budibase

server {
    server_name aaaaa.bbbbb.cc;

    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $http_host;
        proxy_http_version 1.1;
        proxy_pass http://localhost:10000;
        fastcgi_buffers 16 16k;
        fastcgi_buffer_size 32k;
    }

    location /socket/ {
      proxy_http_version  1.1;
      proxy_set_header    Upgrade     $http_upgrade;
      proxy_set_header    Connection  'upgrade';
      proxy_set_header    Host        $host;
      proxy_cache_bypass  $http_upgrade;
      proxy_pass          http://localhost:10000; # Change to your upstream URL
    }

    client_max_body_size 50m;

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/aaaaa.bbbbb.cc/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/aaaaa.bbbbb.cc/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}


server {
    if ($host = aaaaa.bbbbb.cc) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    listen 80;
    server_name aaaaa.bbbbb.cc;
    return 404; # managed by Certbot

}
```


Please, let me know if there's anything else that can be done to get rid of the errors above..   Thanks again for speedy help.

**Reply by aptkingston** [reply]
The top error in that latest message is once again just one of your chrome extensions.

The rest look like more websocket issues, but if hot reloading is working with the latest proxy changes then I wouldn't worry about it. Your nginx config looks good to me. Well, you could remove the `fastcgi` lines since they won't make a difference for Budibase but your `/socket/` config looks good!

In general it's very normal to have some errors and warnings in the browser console when using any site, so unless you're experiencing an actual issue because of them then you don't need to worry and can safely ignore them.

**Reply by marian001** [reply]
Thanks, @aptkingston . After the last restart of nginx service there are no more errors. So I can go work on the fun stuff now  :-)
